File: c/udemy/7-matrizes/matriz_geral.c
// Ler uma matriz quadrada de ordem N (máximo = 10), contendo números reais. Em seguida, fazer as seguintes ações:
// a) calcular e imprimir a soma de todos os elementos positivos da matriz.
// b) fazer a leitura do índice de uma linha da matriz e, daí, imprimir todos os elementos desta linha.
// c) fazer a leitura do índice de uma coluna da matriz e, daí, imprimir todos os elementos desta coluna.
// d) imprimir os elementos da diagonal principal da matriz.
// e) alterar a matriz elevando ao quadrado todos os números negativos da mesma. Em seguida imprimir a matriz alterada.

#include <stdio.h>
#include <math.h>

int main(){
    
    int n, indice;

    printf("Ordem da matriz (max 10): ");
    scanf("%d", &n);

    double mat[n][n], soma_positivos = 0.0;

    // entrada de dados
    for (int i=0; i<n; i++){
        for (int j=0; j<n; j++){
            printf("Elemento [%d][%d]: ", i, j);
            scanf("%lf", &mat[i][j]);
        }
    }

    // imprimir a matriz
    printf("\nMatriz digitada:\n");
    for (int i=0; i<n; i++){
        for (int j=0; j<n; j++){
            printf("%.1lf ", mat[i][j]);
        }
        printf("\n");
    }

    // soma dos positivos
    for (int i=0; i<n; i++){
        for (int j=0; j<n; j++){
            if (mat[i][j] > 0){
                soma_positivos += mat[i][j];
            }
        }
    }
    printf("\nSoma dos positivos = %.1lf\n", soma_positivos);

    // ler índice e imprimir linha
    printf("\nEscolha uma linha (primeira = 0): ");
    scanf("%d", &indice);

    printf("Linha escolhida: ");
    for (int i=0; i<n; i++){
        printf("%.1lf ", mat[indice][i]);
    }
    printf("\n");

    // ler índice e imprimir coluna
    printf("\nEscolha uma coluna (primeira = 0): ");
    scanf("%d", &indice);

    printf("Coluna escolhida: ");
    for (int i=0; i<n; i++){
        printf("%.1lf ", mat[i][indice]);
    }
    printf("\n");

    // diagonal principal
    printf("\nDiagonal principal: ");
    for (int i=0; i<n; i++){
        for (int j=0; j<n; j++){
            if (i == j){
                printf("%.1lf ", mat[i][j]);
            }
        }
    }
    printf("\n");

    // matriz alterada
    printf("\nMatriz alterada:\n");
    for (int i=0; i<n; i++){
        for (int j=0; j<n; j++){
            if (mat[i][j] < 0){
                mat[i][j] = pow(mat[i][j], 2);
            }
        }
    }

    for (int i=0; i<n; i++){
        for (int j=0; j<n; j++){
            printf("%.2lf ", mat[i][j]);
        }
        printf("\n");
    }
    printf("\n");

    return 0;
}

File: python/udemy/7-matrizes/matriz_geral.py
# Ler uma matriz quadrada de ordem N (máximo = 10), contendo números reais. Em seguida, fazer as seguintes ações: 
# a) calcular e imprimir a soma de todos os elementos positivos da matriz. 
# b) fazer a leitura do índice de uma linha da matriz e, daí, imprimir todos os elementos desta linha. 
# c) fazer a leitura do índice de uma coluna da matriz e, daí, imprimir todos os elementos desta coluna. 
# d) imprimir os elementos da diagonal principal da matriz. 
# e) alterar a matriz elevando ao quadrado todos os números negativos da mesma. Em seguida imprimir a matriz alterada.

n = int(input('Ordem da matriz (máximo 10): '))
mat = [[] for _ in range(n)]
mat_alt = [[] for _ in range(n)]
positivos = 0

for i in range(n):
    for j in range(n):
        mat[i].append(float(input(f'Elemento [{i}][{j}]: ')))

# soma positivos
for i in range(n):
    for j in range(n):
        if mat[i][j] > 0:
            positivos += mat[i][j]

print(f'\nSoma dos elementos positivos: {positivos}')

# elementos de uma linha
r = int(input('\nLinha da matriz: '))
if 0 < r <= n:
    for ele_c in mat[r]:
        print(ele_c)
else:
    print(f'A linha {r} não existe na matriz!')

# elementos de uma coluna
c = int(input('\nColuna da matriz: '))
if 0 < c <= n:
    for i in range(n):
        print(mat[i][c])
else:
    print(f'A coluna {c} não existe na matriz!')

# elementos da diagonal principal
print('\nElementos da diagonal principal: ')
for i in range(n):
    for j in range(n):
        if i == j:
            print(mat[i][j])

# matriz com elementos negativos ao quadrado
mat_alt = mat
for i in range(n):
    for j in range(n):
        if mat_alt[i][j] < 0:
            mat_alt[i][j] = mat_alt[i][j] ** 2
print(f'\nMatriz alterada: {mat_alt}')